clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -96,6 +97,10 @@
     private final Metrics metrics;
     private final long defaultApiTimeoutMs;
 
+    private final AtomicReference<CompletableFuture> activeFutures = new AtomicReference<>();
+    private AtomicBoolean shouldWakeup = new AtomicBoolean(false);
+    private volatile boolean closed = false;
+
```

Comment:
it looks better to have initialization in constructor but that's not a thing of first importance.